db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2169 @@
+    try(final var statement = connection.createStatement()) {
+      // create the source type
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_source_type
+          VALUES ('%s');
+          """.formatted(source_type_name)
+      );
+    }
+  }
+
+  protected void insertDerivationGroup(String derivation_group_name, String source_type_name) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the derivation group
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.derivation_group
+          VALUES ('%s', '%s');
+          """.formatted(derivation_group_name, source_type_name)
+      );
+    }
+  }
+
+  protected void insertExternalSource(ExternalSource externalSource) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      System.out.println("STARTING " + externalSource);
+      // create the source
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_source
+          VALUES ('%s', '%s', '%s', '%s', '%s', '%s', '%s');
+          """.formatted(
+              externalSource.key,
+              externalSource.source_type_name,
+              externalSource.derivation_group_name,
+              externalSource.valid_at,
+              externalSource.start_time,
+              externalSource.end_time,
+              externalSource.created_at
+          )
+      );
+      System.out.println("FINISHED " + externalSource);
+    }
+  }
+
+  protected void insertExternalEvent(ExternalEvent externalEvent) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the event
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_event
+          VALUES ('%s', '%s', '%s', '%s', '%s', '%s');
+          """.formatted(
+              externalEvent.key,
+              externalEvent.event_type_name,
+              externalEvent.source_key,
+              externalEvent.derivation_group_name,
+              externalEvent.start_time,
+              externalEvent.duration
+          )
+      );
+    }
+  }
+
+  protected void associateDerivationGroupWithPlan(int planId, String derivationGroupName) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the event type
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.plan_derivation_group
+          VALUES ('%s', '%s');
+          """.formatted(planId, derivationGroupName)
+      );
+    }
+  }
+
+  /**
+   * Get all derived events.
+   */
+  protected List<DerivedEvent> getDerivedEvents(String end) throws SQLException {
```

Comment:
The `String end` parameter should be removed from this method.

All but three usages pass `ORDER BY source_key`, and in **no usages** are you actually **using** the given order in your checks (`containsAll` does not check order, just presence)
- For the one usage that provides`where derivation_group=%DG`, it is preferable to split that into its own override (`getDerivedEvents(String dg_name)`)

Reply:
Okay! Do you have any preference on how/where we declare that override?

Reply:
Right after the base method